rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,923 @@
+
+#### Cloud-friendly pricing model
+
+If the given system has a cost, the pricing model should be cloud-friendly and/or usage-based.
+
+A per-agent or per-build model based on peak usage in a month is not a good model, because our peak build times are generally short-lived (e.g. around feature freeze).
+
+A model based on build-minutes can also be bad, if it encourages running things in parallel on bigger machines to keep costs down. For example, running two tasks on a single 2-CPU machine with our own orchestration should not be cheaper than running two tasks on two 1-CPU machines using the system's built-in orchestration.
+
+#### Public access
+
+Kibana is a publicly-available repository with contributors from outside Elastic. CI information needs to be available publicly in some form.
+
+#### Secrets handling
+
+Good, first-class support for handling secrets is a must-have for any CI system. This support can take many forms.
+
+- Secrets should not need to be stored in plaintext, in a repo nor on the server.
+- For systems provided as a service, it is ideal if secrets are kept mostly/entirely on our infrastructure.
+- There should be protections against accidentally leaking secrets to the console.
+- There should be programmatic ways to manage secrets.
+- Secrets are, by nature, harder to handle. However, the easier the system makes it, the more likely people are to follow best practices.
+
+#### Support or Documentation
+
+For paid systems, both self-hosted and as a service, good support is important. If a problem specific to Elastic is causing us downtime, we expect quick and efficient support. Again, 100s of developers are potentially affected by downtime.
+
+For open source solutions, good documentation is especially important. If much of the operational knowledge of a system can only be gained by working with the system and/or reading the source code, it will be harder to solve problems quickly.
+
+#### Scheduled Builds
+
+We have certain pipelines (ES Snapshots) that run once daily, and `master` CI currently only runs once an hour. We need the ability to configure scheduled builds.
+
+#### Container support
+
+We have the desire to use containers to create fast, clean environments for CI stages that can also be used locally. We think that we can utilize [modern layer-caching options](https://github.com/moby/buildkit#cache), both local and remote, to optimize bootstrapping various CI stages, doing retries, etc.
+
+For self-hosted options, containers will allow us to utilize longer-running instances (with cached layers, git repos, etc) without worrying about polluting the build environment between builds.
+
+If we use containers for CI stages, when a test fails, developers can pull the image and reproduce the failure in the same environment that was used in CI.
+
+So, we need a solution that at least allows us to build and run our own containers. The more features that exist for managing this, the easier it will be.
```

Comment:
From my skimming, Concourse looks like a build tool - in which we [have chosen](https://github.com/elastic/kibana/blob/master/rfcs/text/0015_bazel.md) to use Bazel for that.

Reply:
I'm not super familiar with Concourse, but I know of it. I didn't really consider it in this round mostly because I've heard not great things about it from people who have used it, and the UI is such a turn-off for me: https://ci.concourse-ci.org/teams/main/pipelines/concourse